### PR TITLE
fix: [CI-9748]: Add gcp profiler logservice

### PR DESCRIPTION
**Pull request by smjt-h**
Added gcp profiler to log service
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54759)

### CODE REVIEW

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/config/config.go [hunk 1]
```diff
@@ -13,9 +13,12 @@ import (
...
 // Config provides the system configuration.
 type Config struct {
-	Debug    bool          `envconfig:"LOG_SERVICE_DEBUG_MODE"`
-	Trace    bool          `envconfig:"LOG_SERVICE_TRACE"`
-	CacheTTL time.Duration `envconfig:"LOG_SERVICE_CACHE_TTL" default:"168h"`
+	Debug          bool          `envconfig:"LOG_SERVICE_DEBUG_MODE"`
+	Trace          bool          `envconfig:"LOG_SERVICE_TRACE"`
+	CacheTTL       time.Duration `envconfig:"LOG_SERVICE_CACHE_TTL" default:"168h"`
+	ServiceName    string        `envconfig:"LOG_SERVICE_NAME" default:"log-service"`
+	EnableProfiler bool          `envconfig:"LOG_SERVICE_PROFILER_ENABLED"`
```
Comment (vistaarjuneja) [inline product/log-service/config/config.go:20]: Keep a separate nested struct for Profiler related envs

File: product/log-service/cli/server/server.go [hunk 2]
```diff
@@ -239,3 +251,18 @@ func initLogging(c config.Config) {
 		l.SetLevel(logrus.TraceLevel)
 	}
 }
+
+func StartProfiler(config *config.Config) error {
+	hostName := ""
```
Comment (vistaarjuneja) [inline product/log-service/cli/server/server.go:256]: remove this line

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/config/config.go (hunk 1 above)
Comment (smjt-h) [reply product/log-service/config/config.go]: done

File: product/log-service/cli/server/server.go (hunk 2 above)
Comment (smjt-h) [reply product/log-service/cli/server/server.go]: done

**Comment by smjt-h** [comment]
trigger uts

File: product/log-service/config/config.go [hunk 3]
```diff
@@ -98,6 +98,12 @@ type Config struct {
 	Zip struct {
 		LIMIT_FILES int `envconfig:"LOG_SERVICE_LIMIT_FILES" default:"100"`
 	}
+
+	Profiler struct {
+		ServiceName    string `envconfig:"LOG_SERVICE_NAME" default:"log-service"`
```
Comment (vistaarjuneja) [inline product/log-service/config/config.go:103]: For anything in this struct prefix profiler in the env variable.
`LOG_SERVICE_PROFILER_SERVICE_NAME` `LOG_SERVICE_PROFILER_ENVIRONMENT`.

Comment (smjt-h) [reply product/log-service/config/config.go]: done

**Comment by smjt-h** [comment]
trigger uts

**Pull request merged**